src/object/cli_obj.c

```diff
@@ -941,6 +943,20 @@ obj_shards_2_fwtgts(struct dc_object *obj, uint32_t map_ver, uint8_t *bit_map,
 						  tgt++, obj_auxi);
 			if (rc != 0)
 				return rc;
+
+			if (update && req_tgts->ort_srv_disp) {
+				struct daos_shard_tgt	*tmp = head;
+				struct daos_shard_tgt	*last = tgt - 1;
+
+				for (; tmp != last; tmp++) {
+					/* Two shards locate on the same target,
+					 * OSA case, will handle it via internal
+					 * transaction.
+					 */
+					if (tmp->st_tgt_id == last->st_tgt_id)
+						return -DER_SHARD_OVERLAP;
```

Comment:
please add some error or warning message (or at least some IO message) to help for debugging. thanks.

Reply:
And also do you think we need check the shard status? i.e. only do this dtx conversion during server addition. Since I am not sure we want to use this generally?

Reply:
I will add debug message when detected this case. As for whether need to check the status, I am not sure whether we will allow to migrate data to the server that is in rebuilding. But anyway, if that happened, the patch does not make the things worse, it does not change the RPC destination, instead, it only changes the RPC format.